doc/source/train/examples/lightning/lightning_cola_advanced.ipynb

```diff
@@ -0,0 +1,465 @@
+    "- Do batch prediction on multiple GPUs, and evaluate the results."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 35,
+   "metadata": {
+    "tags": [
+     "remove-cell"
+    ]
+   },
+   "outputs": [],
+   "source": [
+    "SMOKE_TEST = True"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 3,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import ray\n",
+    "import torch\n",
+    "import pytorch_lightning as pl\n",
+    "import torch.nn.functional as F\n",
+    "from torch.utils.data import DataLoader, random_split\n",
+    "from transformers import AutoTokenizer, AutoModelForSequenceClassification\n",
+    "from datasets import load_dataset, load_metric\n",
+    "import numpy as np"
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## 1. Pre-process CoLA Dataset\n",
+    "\n",
+    "CoLA is a binary sentence classification task with 10.6K training examples. First, we download the dataset and metrics using the HuggingFace API, and create Ray Datasets for each split accordingly."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 4,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stderr",
+     "output_type": "stream",
+     "text": [
+      "Reusing dataset glue (/home/ray/.cache/huggingface/datasets/glue/cola/1.0.0/dacbe3125aa31d7f70367a07a8a9e72a5a0bfeb5fc42e75c9db75b96da6053ad)\n",
+      "100%|██████████| 3/3 [00:00<00:00, 948.44it/s]\n"
+     ]
+    }
+   ],
+   "source": [
+    "dataset = load_dataset(\"glue\", \"cola\")\n",
+    "metric = load_metric(\"glue\", \"cola\")"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "ray_datasets = ray.data.from_huggingface(dataset)"
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Next, define a preprocessor that tokenizes the input sentences and pads the ID sequence to length 128 using the bert-base-uncased tokenizer. The preprocessor transforms all datasets that we provide to the LightningTrainer later."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 6,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from ray.data.preprocessors import BatchMapper\n",
+    "\n",
+    "tokenizer = AutoTokenizer.from_pretrained(\"bert-base-cased\")\n",
+    "\n",
+    "\n",
+    "def tokenize_sentence(batch):\n",
+    "    encoded_sent = tokenizer(\n",
+    "        batch[\"sentence\"].tolist(),\n",
+    "        max_length=128,\n",
+    "        truncation=True,\n",
+    "        padding=\"max_length\",\n",
+    "        return_tensors=\"pt\",\n",
+    "    )\n",
+    "    batch[\"input_ids\"] = encoded_sent[\"input_ids\"].numpy()\n",
+    "    batch[\"attention_mask\"] = encoded_sent[\"attention_mask\"].numpy()\n",
+    "    batch[\"label\"] = np.array(batch[\"label\"])\n",
```

Comment:
Out of curiosity, what happens if we don't include this conversion?

Reply:
Ray dataset is kind of rigid, which only support data format as pandas or numpy. However the output of tokenizer is pytorch tensors, which will throw error if no conversion.